app/components/header/header.component.html

```diff
@@ -439,3 +427,50 @@ <h5 type="text" id="stashMsgText" class="stashMsgModalComponents"> Stash Message
     </div><!-- /.modal-content -->
   </div><!-- /.modal-dialog -->
 </div>
+
+<!--Modal for adding remote upstream repository-->
+<div id="set-upstream-modal" class="modal fade" tabindex="-1" role="dialog" aria-labelledby="mySmallModalLabel"
+  aria-hidden="true">
+  <div class="modal-dialog modal-sm">
+    <div class="modal-content" style=" width: 602px !important">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true" onclick="clearUpstreamModalText()" class="upstreamModalComponents">&times;</span>
+        </button>
+        <h4 class="modal-title">Configure an Upstream Repository</h4>
+      </div>
+      <div class="modal-body" id="modal-text-box">
+        <h5 type="text">Upstream:</h5>
+        <p id="display-upstream"></p>
+        <button type="button" class="btn btn-danger btn-sm" id="deleteUpstreamBtn" data-dismiss="modal" onclick="showUpstreamDelete()">Delete</button>
+      </div>
+      <div class="modal-body" id="modal-text-box">
+        Please provide the HTTP path to the original repository:
+        <input type="text" id="upstream-path" class="form-control" style=" width: 554px !important" placeholder="https://github.com/ORIGINAL_OWNER/ORIGINAL_OWNER_REPOSITORY.git">
+      </div>
+      <div class="modal-footer">
+        <button type="button" class="btn btn-primary" data-dismiss="modal" id="setUpstreamButton" onclick="setUpstreamRepo()">Confirm</button>
+      </div>
+    </div>
+  </div>
+</div>
+
+<!-- Confirmation Modal for deleting upstream -->
+<div id="delete-upstream-modal" class="modal fade">
+  <!-- <input type="hidden" id="branch-to-delete" name="branch-to-delete" /> -->
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+        <h4 class="modal-title">Confirmation</h4>
+      </div>
+      <div class="modal-body">
+        Are you sure you want to delete the upstream repository?
+      </div>
+      <div class="modal-footer">
+        <button type="button" class="btn btn-default" data-dismiss="modal">Close</button>
+        <button type="button" class="btn btn-warning" id="localDeleteButton" data-dismiss="modal" onClick="deleteUpstream()">Delete</button>
```

Comment:
Should this be `btn-danger` to match the delete button on the previous page?

Reply:
Do you have a repository open when it looks like this?

Reply:
Yeah, If I highlight the text it shows up properly! (Same case with a saved upstream)
![image](https://user-images.githubusercontent.com/50643496/59719327-f881ec00-91d0-11e9-8b73-15ee38d0ee2f.png)

Reply:
Are you on Windows? Interesting....

Reply:
Yeah I'm on windows! Seems like what's happening is in color.ts it finds every <p> element and colors it depending on the style (for default this is white). Not sure why it isn't happening on Mac, I think you could probably fix it by changing the <p> tag to something else or adding a colour with the !important modifer to the CSS

Reply:
I took your suggestion and used the !important tag in the CSS and also changed the button to `btn-danger` , please see the latest update!

Reply:
Works for me! Nice work!